pkg/s3select/select.go

```diff
@@ -338,24 +356,29 @@ func (s3Select *S3Select) Evaluate(w http.ResponseWriter) {
 	var inputRecord sql.Record
 	var outputRecord sql.Record
 	var err error
-	var data []byte
+	bufW := bufio.NewWriter(ioutil.Discard)
 	sendRecord := func() bool {
 		if outputRecord == nil {
 			return true
 		}
 
-		if data, err = s3Select.marshal(outputRecord); err != nil {
+		b := bufPool.Get().(*bytes.Buffer)
```

Comment:
A wrong use of pool handling here and there is bug that you are not putting the buffer back on success case.  You would need to use like below
```go
buf := bufPool.Get().(*bytes.Buffer)
defer func() {
    buf.Reset()
    bufPool.Put(buf)
}()
```

Reply:
On success case, buf is putted back at here https://github.com/yaozongyou/minio/blob/speed-up-select-on-csv/pkg/s3select/message.go#L326

the response message is sended at a different go routine transfered by a channel.

Reply:
Got it